### PR TITLE
[Enhancement] Add type hints for mmcv/arraymisc and mmcv/video

### PR DESCRIPTION
**Pull request by triple-Mu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add type hints for mmcv/arraymisc/quantization.py and mmcv/video/optflow.py 

## Modification

Only the above two files were modified

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/arraymisc/quantization.py [hunk 1]
```diff
@@ -29,13 +35,17 @@ def quantize(arr, min_val, max_val, levels, dtype=np.int64):
     return quantized_arr
 
 
-def dequantize(arr, min_val, max_val, levels, dtype=np.float64):
+def dequantize(arr: np.ndarray,
+               min_val: Union[int, float],
+               max_val: Union[int, float],
+               levels: int,
+               dtype: np.type = np.float64) -> tuple:
```
Comment (zhouzaida) [inline mmcv/arraymisc/quantization.py:42]: 'numpy' has no attribute 'type'

Comment (zhouzaida) [inline mmcv/arraymisc/quantization.py:42]:
```suggestion
               dtype=np.float64) -> tuple:
```

File: mmcv/tensorrt/tensorrt_utils.py [hunk 2]
```diff
@@ -8,12 +9,12 @@
 from .preprocess import preprocess_onnx
 
 
-def onnx2trt(onnx_model,
-             opt_shape_dict,
-             log_level=trt.Logger.ERROR,
-             fp16_mode=False,
-             max_workspace_size=0,
-             device_id=0):
+def onnx2trt(onnx_model: onnx.ModelProto,
```
Comment (zhouzaida) [inline mmcv/tensorrt/tensorrt_utils.py:12]:
```suggestion
def onnx2trt(onnx_model: Union[str, onnx.ModelProto],
```

File: mmcv/arraymisc/quantization.py (hunk 1 above)
Comment (triple-Mu) [reply mmcv/arraymisc/quantization.py]: So what is the `np.float64` type? When I try  `type(np.int64)` it returns `<class 'type'>`, so is this type hint `type` ?

Comment (zhouzaida) [reply mmcv/arraymisc/quantization.py]: np.float64 is a class and its type is metaclass `type` so we can ignore the type hint for `dtype`.

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**